### ISSUE TITLE
Index out of range crash

### ISSUE THREAD

**Issue by medikoo** [tj/punt#10]
I use punt in long living processes, and already twice I experienced following crashes:

```
buffer.js:506
    throw new RangeError('index out of range');
          ^
RangeError: index out of range
    at checkOffset (buffer.js:506:11)
    at Buffer.readUInt32BE (buffer.js:580:5)
    at Object.module.exports [as decode] (/home/eregistrations/els/node_modules/punt/node_modules/amp-message/node_modules/amp/lib/decode.js:21:19)
    at decode (/home/eregistrations/els/node_modules/punt/node_modules/amp-message/index.js:80:18)
    at new Message (/home/eregistrations/els/node_modules/punt/node_modules/amp-message/index.js:35:37)
    at Server.onmessage (/home/eregistrations/els/node_modules/punt/lib/server.js:63:13)
    at Socket.emit (events.js:110:17)
    at UDP.onMessage (dgram.js:472:8)

npm ERR! Linux 3.13.0-36-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v0.12.7
npm ERR! npm  v2.11.3
npm ERR! code ELIFECYCLE
npm ERR! eregistrations-salvador@0.2.0 start: `node bin/setup && node node_modules/mano/bin/start --quick`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the eregistrations-salvador@0.2.0 start script 'node bin/setup && node node_modules/mano/bin/start --quick'.
npm ERR! This is most likely a problem with the eregistrations-salvador package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node bin/setup && node node_modules/mano/bin/start --quick
npm ERR! You can get their info via:
npm ERR!     npm owner ls eregistrations-salvador
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/eregistrations/els/npm-debug.log
```

Message that's source of the issue is unknown to me, as at given point there certainly was no cross process communication involved in my setup.

It'll be great if punt internally catches such errors and emits them through error messages, otherwise whole process gets down.

**Comment by Qard** [comment]
It seems your server is receiving packets that aren't amp messages. Do you have non-punt clients connecting to the server for some reason?

Regardless, I agree that the server should expect parse errors and properly emit error events. I'll look into that tonight.

**Comment by medikoo** [comment]
> Do you have non-punt clients connecting to the server for some reason?

No, technically I have no clue what's the source of that errorneous message. I hard coded try/catch on my side, so maybe on next event log will provide some hint

**Comment by Qard** [comment]
I've released a 2.3.0 version that includes a `parse-error` event which receives the error and failed buffer. Note that it is `parse-error` and not `error` as the `error` event in node is a special-cased event name that will throw when not listened to.

That should be sufficient as an immediate fix, and a better way to inspect the messages which are causing this failure.

@tj Do you think this event should be renamed to `error` and pushed as a 3.0.0? I'm not sure if I think punishing developers for not handling parse errors is the right thing to do here.

**Comment by medikoo** [comment]
@Qard great thanks! That solves the issue to me
